opentreemap/stormwater/migrations/0007_drainage_area_permissions.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.db import migrations
+
+
+def add_permissions_all(apps, schema_editor):
+    Instance = apps.get_model('treemap', 'Instance')
+    Role = apps.get_model('treemap', 'Role')
+    FieldPermission = apps.get_model('treemap', 'FieldPermission')
+    need_to_clear_cache = False
+
+    def update_field_permissions(perm_specs, role, instance):
+        existing = FieldPermission.objects.filter(
+            role=role, instance=instance)
+        if existing.exists():
+            for perm in perm_specs:
+                perm['defaults'] = {
+                    'permission_level': role.default_permission}
+                FieldPermission.objects.get_or_create(**perm)
+            return False
+        else:
+            perm_specs = [FieldPermission(**perm) for perm in perm_specs]
+            for perm in perm_specs:
+                perm.permission_level = role.default_permission
+            FieldPermission.objects.bulk_create(perm_specs)
+            return True
+
+    for instance in Instance.objects.filter(config__contains='"Plot",'):
+        types = instance.config['map_feature_types']
+        if 'Bioswale' in types or 'RainGarden' in types:
+            for role in Role.objects.filter(instance=instance):
+                perm_specs = [
+                    {'model_name': 'Bioswale',
+                     'field_name': 'drainage_area',
+                     'role': role,
+                     'instance': instance},
+                    {'model_name': 'RainGarden',
+                     'field_name': 'drainage_area',
+                     'role': role,
+                     'instance': instance}]
+                need_to_clear_cache = \
+                    update_field_permissions(perm_specs, role, instance) or \
+                    need_to_clear_cache
+
+    if need_to_clear_cache:
+        from object_caches import clear_caches
+        clear_caches()
```

Comment:
Well we're looping over the `Role`s anyways to update field permissions.... but yes, it would definitely be fine to just increment `adjuncts_timestamp` on any `Instance` with Bioswales or Rain Gardens - the only problem with incrementing the timestamp too much would be performance concerns, and there aren't many instances with those map features enabled.